### PR TITLE
Improve the API documentation and markdown files

### PR DESCRIPTION
**Pull request by aashikam**
## Purpose
$title
https://github.com/ballerina-platform/ballerina-standard-library/issues/1383

### CODE REVIEW

**Review by MadhukaHarith92 — APPROVED** [review]

File: rabbitmq-ballerina/Module.md [hunk 1]
```diff
@@ -0,0 +1,178 @@
+## Overview
+
+This module provides the capability to send and receive messages by connecting to the RabbitMQ server.
+
+RabbitMQ gives your applications a common platform to send and receive messages, and your messages a safe place to live until received. RabbitMQ is one of the most popular open source message brokers. It is lightweight and easy to deploy on premises and in the cloud.
```
Comment (praneesha) [inline rabbitmq-ballerina/Module.md:5]:
```suggestion
RabbitMQ gives your applications a common platform to send and receive messages and a safe place for your messages to live until received. RabbitMQ is one of the most popular open-source message brokers. It is lightweight and easy to deploy on-premise and in the cloud.
```

File: rabbitmq-ballerina/Module.md [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+        password: "password"
+    };
+    rabbitmq:Client rabbitmqClient = check new("localhost", 5672, configs);
+```
+
+The `rabbitmq:Client` can now be used to send and receive messages as described in the subsequent sections.
+
+#### Using Exchanges and Queues
+
+Client applications work with exchanges and queues, which are the high-level building blocks of the AMQP protocol. These must be declared before they can be used. The following code declares an exchange, and a server-named queue and then binds them together.
```
Comment (praneesha) [inline rabbitmq-ballerina/Module.md:37]:
```suggestion
Client applications work with exchanges and queues, which are the high-level building blocks of the AMQP protocol. These must be declared before they can be used. The following code declares an exchange and a server-named queue and then binds them together.
```

File: rabbitmq-ballerina/Module.md [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+    check rabbitmqClient->exchangeDeclare("MyExchange", rabbitmq:DIRECT_EXCHANGE);
+    check rabbitmqClient->queueDeclare("MyQueue", { durable: true,
+                                                   exclusive: false,
+                                                   autoDelete: false });
+    check rabbitmqClient->queueBind("MyQueue", "MyExchange", "routing-key");
+```
+
+This sample code will declare,
+- a durable auto-delete exchange of the type `rabbitmq:DIRECT_EXCHANGE`
+- a durable, non-exclusive non-auto-delete queue with a well-known name
```
Comment (praneesha) [inline rabbitmq-ballerina/Module.md:61]:
```suggestion
- a durable, non-exclusive, non-auto-delete queue with a well-known name
```

File: rabbitmq-ballerina/Module.md [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+```
+
+The `rabbitmq:Message` record received can be used to retrieve its contents.
+
+### Advanced Usage
+
+#### Client Acknowledgements
+
+The message consuming is supported by mainly two types of acknowledgement modes, which are auto acknowledgements and client acknowledgements.
+Client acknowledgements can further be divided into to two different types as positive and negative acknowledgements.
```
Comment (praneesha) [inline rabbitmq-ballerina/Module.md:144]:
```suggestion
Client acknowledgements can further be divided into two different types as positive and negative acknowledgements.
```

File: rabbitmq-ballerina/Module.md [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+The default acknowledgement mode is auto-ack (messages are acknowledged immediately after consuming). Following examples show the usage of positive and negative acknowledgements.
```
Comment (praneesha) [inline rabbitmq-ballerina/Module.md:145]:
```suggestion
The default acknowledgement mode is auto-ack (messages are acknowledged immediately after consuming). The nfollowing examples show the usage of positive and negative acknowledgements.
```

File: rabbitmq-ballerina/Module.md [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+        autoAck: false
+    }
+    service rabbitmq:Service on channelListener {
+        remote function onMessage(rabbitmq:Message message) {
+            rabbitmq:Error? result = caller->basicNack(true, requeue = false);
+        }
+    }
+```
+
+The negatively-acknowledged (rejected) messages can be re-queued byt setting `requeue` to `true`.
```
Comment (praneesha) [inline rabbitmq-ballerina/Module.md:178]:
```suggestion
The negatively-acknowledged (rejected) messages can be re-queued by setting the `requeue` to `true`.
```

File: rabbitmq-ballerina/Package.md [hunk 7]
```diff
@@ -1,137 +1,16 @@
...
 
-[RabbitMQ](https://www.rabbitmq.com/) is one of the most popular open-source enterprise messaging systems modelled on the Advanced Message Queuing Protocol (AMQP) standard. This guide covers the Ballerina RabbitMQ package and its public API. 
- 
-Key sections include: 
-- Connecting to RabbitMQ
-- Using Exchanges and Queues
-- Publishing Messages
-- Consuming Messages Using Consumer Services
-- Client Acknowledgements 
+This package provides the capability to send and receive messages by connecting to the RabbitMQ server. RabbitMQ gives your applications a common platform to send and receive messages, and your messages a safe place to live until received. RabbitMQ is one of the most popular open source message brokers. It is lightweight and easy to deploy on premises and in the cloud.
```
Comment (praneesha) [inline rabbitmq-ballerina/Package.md:6]:
```suggestion
This package provides the capability to send and receive messages by connecting to the RabbitMQ server. RabbitMQ gives your applications a common platform to send and receive messages and a safe place for your messages to live until received. RabbitMQ is one of the most popular open-source message brokers. It is lightweight and easy to deploy on-premise and in the cloud. 
```

File: rabbitmq-ballerina/caller.bal [hunk 8]
```diff
@@ -37,10 +37,10 @@ public client class Caller {
 
     # Rejects one or several received messages.
     # ```ballerina
-    # rabbitmq:Error? nackResult = caller->basicNack(true, requeue = false);
+    # check caller->basicNack(true, requeue = false);
     # ```
     #
-    # + multiple - `true` to reject all messages up to and including the called on message and
+    # + multiple - Set to `true` to reject all messages, up to and including the called on message and
```
Comment (praneesha) [inline rabbitmq-ballerina/caller.bal:43]:
```suggestion
    # + multiple - Set to `true` to reject all messages up to and including the called on message and
```

File: rabbitmq-ballerina/client.bal [hunk 9]
```diff
@@ -40,106 +43,107 @@ public client class Client {
...
+    # + name - The name of the queue
+    # + config - The configurations required to declare a queue
     # + return - `()` if the queue was successfully generated or else a `rabbitmq:Error`
-    #               if an I/O error is encountered
+    #               if an I/O error occurred
     isolated remote function queueDeclare(string name, QueueConfig? config = ()) returns Error? {
         return nativeQueueDeclare(name, config, self.amqpChannel);
     }
 
     # Declares a queue with a server generated name.
```
Comment (praneesha) [inline rabbitmq-ballerina/client.bal:57]:
```suggestion
    # Declares a queue with a server-generated name.
```

File: rabbitmq-ballerina/client.bal [hunk 10]
```diff
@@ -40,106 +43,107 @@ public client class Client {
...
     # ```ballerina
-    # string|rabbitmq:Error? queueName = newClient->queueAutoGenerate();
+    # string queueName = check rabbitmqClient->queueAutoGenerate();
     # ```
     #
     # + return - The name of the queue, or else a `rabbitmq:Error`
```
Comment (praneesha) [inline rabbitmq-ballerina/client.bal:62]:
```suggestion
    # + return - The name of the queue or else a `rabbitmq:Error`
```

File: rabbitmq-ballerina/client.bal [hunk 11]
```diff
@@ -149,57 +153,57 @@ public client class Client {
...
         return nativeQueuePurge(queueName, self.amqpChannel);
     }
 
     # Closes the `rabbitmq:Client`.
     # ```ballerina
-    # rabbitmq:Error? closeResult = newClient.close();
+    # check rabbitmqClient.close();
     # ```
     #
     # + closeCode - The close code (for information, go to the "Reply Codes" section in the
```
Comment (praneesha) [inline rabbitmq-ballerina/client.bal:195]:
```suggestion
    # + closeCode - The close code (for information, go to the [Reply Codes]
```

File: rabbitmq-ballerina/client.bal [hunk 12]
```diff
@@ -149,57 +153,57 @@ public client class Client {
...
     #               [AMQP 0-9-1 specification] (#https://www.rabbitmq.com/resources/specs/amqp0-9-1.pdf))
```
Comment (praneesha) [inline rabbitmq-ballerina/client.bal:196]:
```suggestion
    #               (#https://www.rabbitmq.com/resources/specs/amqp0-9-1.pdf))
```
Please check if the link is added correctly.

File: rabbitmq-ballerina/client.bal [hunk 13]
```diff
@@ -149,57 +153,57 @@ public client class Client {
...
     }
 
     # Aborts the RabbitMQ `rabbitmq:Client`. Forces the `rabbitmq:Client` to close and waits for all the close operations
     # to complete. Any encountered exceptions in the close operations are discarded silently.
     # ```ballerina
-    # rabbitmq:Error? abortResult = newClient.abort(320, "Client Aborted");
+    # check rabbitmqClient.abort(320, "Client Aborted");
     # ```
     #
     # + closeCode - The close code (for information, go to the "Reply Codes" section in the
```
Comment (praneesha) [inline rabbitmq-ballerina/client.bal:209]:
```suggestion
    # + closeCode - The close code (for information, go to the "Reply Codes" section in the
```
Same as above.

File: rabbitmq-ballerina/listener.bal [hunk 14]
```diff
@@ -77,16 +92,19 @@ public class Listener {
...
+    # ```
     #
     # + return - `()` or else  a `rabbitmq:Error` upon failure to close ChannelListener.
     public isolated function immediateStop() returns error? {
         return abortConnection(self);
     }
 }
 
-# Configurations required to create a subscription.
+# Configurations required to creating a subscription.
```
Comment (praneesha) [inline rabbitmq-ballerina/listener.bal:105]:
```suggestion
# Configurations required to create a subscription.
```

File: rabbitmq-ballerina/message.bal [hunk 15]
```diff
@@ -16,7 +16,7 @@
 
 # Represents the message, which a RabbitMQ server sends to its subscribed services.
 #
-# + content - Content of the message
+# + content - The content of the message
 # + routingKey - The routing key to which the message is sent to
```
Comment (praneesha) [inline rabbitmq-ballerina/message.bal:20]:
```suggestion
# + routingKey - The routing key to which the message is sent 
```

File: rabbitmq-ballerina/message.bal [hunk 16]
```diff
@@ -16,7 +16,7 @@
...
 # + exchange - The exchange to which the message is sent to
```
Comment (praneesha) [inline rabbitmq-ballerina/message.bal:21]:
```suggestion
# + exchange - The exchange to which the message is sent 
```

File: rabbitmq-ballerina/rabbitmq_commons.bal [hunk 17]
```diff
@@ -52,9 +52,9 @@ public type BasicProperties record {|
 
 # Additional configurations used to declare a queue.
 #
-# + durable - True if declaring a durable queue (the queue will survive in a server restart)
-# + exclusive - True if declaring an exclusive queue (restricted to this connection)
-# + autoDelete - True if declaring an auto-delete queue (the server will delete it when it is no longer in use)
+# + durable - Set to true if declaring a durable queue
+# + exclusive - Set to true if declaring an exclusive queue
+# + autoDelete - Set to true if declaring an auto-delete queue
 # + arguments - Other properties (construction arguments) for the queue
```
Comment (praneesha) [inline rabbitmq-ballerina/rabbitmq_commons.bal:58]:
```suggestion
# + arguments - Other properties (construction arguments) of the queue
```

File: rabbitmq-ballerina/rabbitmq_commons.bal [hunk 18]
```diff
@@ -65,16 +65,16 @@ public type QueueConfig record {|
 
 # Additional configurations used to declare an exchange.
 #
-# + durable - True if declaring a durable exchange (the exchange will survive in a server restart)
-# + autoDelete - True if an autodelete exchange is declared (the server will delete it when it is no longer in use)
+# + durable - Set to true if declaring a durable exchange
```
Comment (praneesha) [inline rabbitmq-ballerina/rabbitmq_commons.bal:68]:
```suggestion
# + durable - Set to `true` if a durable exchange is declared
```

File: rabbitmq-ballerina/rabbitmq_commons.bal [hunk 19]
```diff
@@ -65,16 +65,16 @@ public type QueueConfig record {|
...
+# + autoDelete - Set to true if an autodelete exchange is declared
```
Comment (praneesha) [inline rabbitmq-ballerina/rabbitmq_commons.bal:69]:
```suggestion
# + autoDelete - Set to `true` if an autodelete exchange is declared
```

File: rabbitmq-ballerina/rabbitmq_commons.bal [hunk 20]
```diff
@@ -99,9 +99,9 @@ public type ConnectionConfiguration record {|
 # QoS settings to limit the number of unacknowledged
 # messages on a channel.
 #
-# + prefetchCount - Maximum number of messages that the server will deliver.
+# + prefetchCount - The maximum number of messages that the server will deliver.
 #                   Give the value as 0 if unlimited
-# + prefetchSize - Maximum amount of content (measured in octets)
+# + prefetchSize - The maximum amount of content (measured in octets)
 #                   that the server will deliver and 0 if unlimited
 # + global - True if the settings should be shared among all consumers
```
Comment (praneesha) [inline rabbitmq-ballerina/rabbitmq_commons.bal:106]:
```suggestion
# + global - `true` if the settings should be shared among all the consumers
```

File: rabbitmq-ballerina/Module.md (hunk 5 above)
Comment (aashikam) [reply rabbitmq-ballerina/Module.md:145]: This suggestion has a type

File: rabbitmq-ballerina/client.bal (hunk 12 above)
Comment (aashikam) [reply rabbitmq-ballerina/client.bal:196]: Link verified

**Review by praneesha — APPROVED** [review]

File: rabbitmq-ballerina/Module.md (hunk 5 above)
Comment (praneesha) [reply rabbitmq-ballerina/Module.md:145]: Corrected. Thanks!

**Pull request merged**